src/pantry-class.js

```diff
@@ -30,6 +35,36 @@ class Pantry {
     }
     return supplyList
   }
+
+  findIngredientIds = (recipe) => {
+    return recipe.requiredIngredients.map(ingredient => ingredient.id)
+  } 
+
+  findIngredientName(id) {
+    if (typeof id === 'number') {
+      let ingredient = ingredientData.find(item => item.id === id);
+      return ingredient.name;
+    }
+  }
+
+  findMissingIngredients = (recipe) => {
+    if (recipe instanceof Recipe === false) {
+      return 'This is not a recipe'
+    }
+
+    let supplyList = this.checkPantryForRecipeIngredients(recipe); 
+    let message = [];
+    
+    recipe.requiredIngredients.forEach(ingredient => {
+      let pantryItem = supplyList.find(item => item.ingredient === ingredient.id);
+      let qtyDifference = pantryItem ? ingredient.amount - pantryItem.amount : ingredient.amount;
+      
+      qtyDifference > 0 ? message.push(`${qtyDifference} ${this.findIngredientName(ingredient.id)}`) : '';
+    });
+    if (message.length > 0) {
+      return `You still need ${message.join(' and ')} to make ${recipe.name}`
+    } 
+  }
```

Comment:
I like how thorough this is - we could probably look at breaking this out into a few different functions.